deep_qa/data/instances/tuple_inference_instance.py

```diff
@@ -0,0 +1,443 @@
+        # Currently, source, time and location aren't being used, but this can be changed in the future.
+
+        return text_list
+
+
+    @classmethod
+    def from_string(cls, input_string: str):
+        """
+        Converts a string representation of the tuple into a TextTuple.
+        Parameters
+        ----------
+        input_string: str
+
+        Returns
+        -------
+        TextTuple
+        """
+        tuple_fields = input_string.lower().split("<>")
+        num_fields = len(tuple_fields)
+        if num_fields < 2:
+            raise ConfigurationError("num_fields (i.e. num tuple fields) not >= 2.  input_string: " + input_string)
+        subject = tuple_fields[0]
+        verb = tuple_fields[1]
+        objects = []
+        context = time = location = source = None
+        for element in tuple_fields[2:]:
+            if element.startswith("context:"):
+                context = element[8:]
+            elif element.startswith("L:"):
+                location = element[2:]
+            elif element.startswith("T:"):
+                time = element[2:]
+            elif element.startswith("source:"):
+                source = element[7:]
+            else:
+                objects.append(element)
+        return TextTuple(subject, verb, objects, context, time, location, source)
+
+
+class TupleInferenceInstance(TextInstance):
+    """
+    A ``TupleInferenceInstance`` is a kind of ``TextInstance`` that has text in multiple slots or phrases
+    (stored as a ``TextTuple``), for both the answer choices and background information.
+    This can be used to store SVO triples.  These tuples are designed to be used in systems which
+    compare question-answer tuples to background knowledge tuples, to preform a sort of "inference" about
+    whether or not the question is answered correctly.
+
+    Parameters
+    ----------
+    answer_tuples: List[List[TextTuple]]
+        This is a list whose length is equal to the num answer options, and each list element corresponds to an
+        answer option.  Each answer has a list of ``TextTuples`` which come from combining that answer option
+        with the question.
+
+    background_tuples: List[TextTuple]
+        This is a list of background ``TextTuples`` (currently used for all answer candidates).
+
+    label: int, default=None
+        The class label (i.e. the index of the correct multiple choice answer) -- corresponds to the
+        indices in self.answer_tuples.
+
+    index: int, default=None
+        The index of the question.
+    """
+
+    def __init__(self, answer_tuples: List[List[TextTuple]], background_tuples: List[TextTuple],
+                 label: int=None, index: int=None):
+        super(TupleInferenceInstance, self).__init__(label, index)
+        self.answer_tuples = answer_tuples
+        self.background_tuples = background_tuples
+        self.object_handling = tuple_inference_parameters['object_handling']
```

Comment:
This should be outside of `__init__`, without the global object.  The class should look like this:

```python
class TupleInferenceInstance(TextInstance):
    """docstring here"""
    object_handling = 'collapse'
    use_context = False

    def __init__(self, answer_tuples: ...):
        ...
```

After that, you can call `self.object_handling` just fine in other methods in this class, and you can change it from a model class by calling `TupleInferenceInstance.object_handling = "something else"`.